admin/scripts/modules/uvod.php

```diff
@@ -98,7 +98,20 @@
 }
 
 if(post('zmenitUdaj')) {
-  dbUpdate('uzivatele_hodnoty', post('udaj'), ['id_uzivatele' => $uPracovni->id()]);
+  if(post('udaj')['login_uzivatele']) {
+    try {
+      dbUpdate('uzivatele_hodnoty', post('udaj'), ['id_uzivatele' => $uPracovni->id()]);
+    }
+    catch (DbException $e) {
+      if (strpos($e, 'Duplicate entry') !== false) {
+        throw new Chyba('Uživatel se stejnou přezdívkou již existuje.');
+      } else {
+        throw new Chyba('Nastal problém s databází: '.$e);
```

Comment:
Výjimky třídy `Chyba` jsou zobrazením pro uživatele a představují nějakou očekávanou chybu, tj. nezalogují se. V tomto případě by ale else větev při normálním fungování neměla nikdy nastat, tj. bylo by lepší jen dál vyhodit původní výjimku (`throw $e`, ovšem to se vyřeší už tím, že catch blok bude chytat jenom specifický typ výjimky a obecná výjimka normálně probublá ven).